client/controller/shared.js

```diff
@@ -35,25 +37,86 @@ export function makeLayoutMiddleware( LayoutComponent ) {
 	};
 }
 
+/**
+ * Loads a a css stylesheet into the page
+ * @param {string} cssUrl - a url to a css resource to be inserted into the page
+ * @param {Function} callback - a callback function to be called when the CSS has been loaded (after 500ms have passed).
+ */
+function loadCSS( cssUrl, callback = noop ) {
+	const link = Object.assign( document.createElement( 'link' ), {
+		rel: 'stylesheet',
+		type: 'text/css',
+		href: cssUrl,
+	} );
+
+	const onload = () => {
+		if ( 'onload' in link ) {
+			link.onload = null;
+		}
+		callback( null, link );
+	};
+
+	if ( 'onload' in link ) {
```

Comment:
Apologies if I'm missing something, but how could `link` possibly have an `onload` property at this point?

Reply:
Do we need that fallback? We only support browsers listed on https://browsehappy.com/. Otherwise I think it would be beneficial to add the information you mention as a comment in the code itself.

Finally we could probably simplify this function a bit:

```javascript
function loadCSS( cssUrl, callback = noop ) {
  const link = Object.assign( document.createElement( 'link' ), {
    rel: 'stylesheet',
    type: 'text/css',
    href: cssUrl,
  } );

  if ( 'onload' in link ) {
    link.onload = () => {
      link.onload = null;

      callback( null, link );
    };
  } else {
    setTimeout( () => {
      callback( null, link );
    }, 500 );
  }

  document.head.appendChild( link );
}
```

Reply:
It's safer to have it then to block loading calypso because of it imo